tests/pipeline/test_resources_evaluation.py

```diff
@@ -217,6 +217,19 @@ async def async_resource1():
     assert len(result) == 13
 
 
+@pytest.mark.parametrize("parallelized", [False, True])
+def test_limit_sync_resource(parallelized: bool) -> None:
+    @dlt.resource(parallelized=parallelized)
+    def sync_resource1():
+        for i in range(1, 10):
+            yield i
+
+    limit = 5
+    result = list(sync_resource1().add_limit(limit))
+    allowed_result_range = range(limit - int(parallelized), limit + 1)
```

Comment:
this is not ideal.
Ideal would be to make the parallel yields exact. I tried this for a few hours to no avail. Next best would be to run this test x times when `parallelized=True` and then ensure that the threshold of yields lies somewhere above 4.5 or so.